src/odemis/acq/stream/_sync.py

```diff
@@ -2038,7 +2039,10 @@ def __init__(self, name, helper_stream, **kwargs):
         self._dwellTime = helper_stream.dwellTime
         self.roi = helper_stream.roi
         self.repetition = helper_stream.repetition
-        self.filename = self._tc_scanner.filename
+        if self._tc_scanner is not None and hasVA(self._tc_scanner, "filename"):
+            self.filename = self._tc_scanner.filename
+        else:
```

Comment:
Just don't create a filename VA at all. That's the proper way to indicate we don't care about the filename.

Also, as `self._tc_scanner` would be a component, it's safe to write just `if self._tc_scanner` instead of `if self._tc_scanner is not None`.